SpatialGDK/Source/SpatialGDK/Private/Interop/WellKnownEntitySystem.cpp

```diff
@@ -130,4 +135,51 @@ void WellKnownEntitySystem::InitializeVirtualWorkerTranslationManager()
 	VirtualWorkerTranslationManager = MakeUnique<SpatialVirtualWorkerTranslationManager>(Receiver, Connection, VirtualWorkerTranslator);
 	VirtualWorkerTranslationManager->SetNumberOfVirtualWorkers(NumberOfWorkers);
 }
+
+void WellKnownEntitySystem::MaybeClaimSnapshotPartition()
+{
+	// Perform a naive leader election where we wait for the correct number of server workers to be present in the deployment, and then
+	// whichever server has the lowest server worker entity id claims the snapshot partition.
+	const Worker_EntityId LocalServerWorkerEntityId = GlobalStateManager->GetLocalServerWorkerEntityId();
+
+	if (LocalServerWorkerEntityId == SpatialConstants::INVALID_ENTITY_ID)
+	{
+		UE_LOG(LogWellKnownEntitySystem, Warning, TEXT("MaybeClaimSnapshotPartition aborted due to lack of local server worker entity"));
+		return;
+	}
+
+	Worker_EntityId LowestEntityId = LocalServerWorkerEntityId;
+
+	int ServerCount = 0;
+	for (const auto& Iter : SubView->GetView())
```

Comment:
iterating through every single entity component is a little scary, and seems avoidable here
Considering this is only called when we first see the relevant component, could we pass through the entity ID (as server worker entity IDs aren't well known- would require modifying the component added) and instead have this function keep track locally how many server worker entities it's seen, as well as the lowest it's seen so far, and on the final one claim the partition if you're still lowest

the approach as is has the benefit of not requiring any bespoke state in the system, as well as implicitly handling the case where servers go down and come back up, although AFAICT we don't care about crash recovery. I think the slight extra complexity is probably worth not iterating through the whole view though

Reply:
+1 to incrementing (or decrementing) something every time we see a worker entity + storing the lowest one. I'd actually argue that's easier to follow too.

Case Component
  if (id < lowest_id) 
     lowest_id = id 
  --servers_required;
  if (servers_required == 0 && lowest_id = my_id)
     claim_partition

Reply:
Will be addressed in future - https://improbableio.atlassian.net/browse/UNR-4500

Reply:
In the future none of this will exist, which might be a reason to disregard comments now, but it seems odd to make follow up tickets to deal with specific comments about code hygiene.

Reply:
We spoke offline- the fact that this doesn't deal with workers crashing was enough (for me at least) to leave it as is for now, as the fix to actually make it crash safe is a little more involved, so if we do end up cleaning this up, we should do that too